client/ios/dropin/Components/EventFormView.swift

```diff
@@ -0,0 +1,237 @@
+                                        byAdding: .minute,
+                                        value: 90,
+                                        to: event.start
+                                    ) ?? event.end
+                                }
+                            }
+                    }
+                } header: {
+                    Text("When?")
+                        .font(.headline)
+                        .foregroundColor(.black)
+                        .textCase(nil)
+                }
+                
+                Section {
+                    VStack {
+                        HStack {
+                            Image(systemName: "magnifyingglass")
+                                .foregroundColor(.gray)
+                            
+                            TextField("Search Map", text: $locationText)
+                                .focused($focusedField, equals: .location)
+                                .onChange(of: locationText) { oldValue, newValue in
+                                    if !newValue.isEmpty {
+                                        updateMapRegion(for: newValue)
+                                    }
+                                }
+                            
+                            Image(systemName: "mic")
+                                .foregroundColor(.gray)
+                        }
+                        .padding(.vertical, 8)
+                        .background(Color(uiColor: .secondarySystemBackground))
+                        .cornerRadius(10)
+                        
+                        Map(initialPosition: .region(region)) {
+                            Marker("Event Location", coordinate: region.center)
+                                .tint(.red)
+                        }
+                        .frame(height: 150)
+                        .cornerRadius(10)
+                        .padding(.top, 8)
+                        // TODO: either properly implement or discard
+                        //.onTapGesture {
+                        //    event.latitude = region.center.latitude
+                        //    event.longitude = region.center.longitude
+                        //    reverseGeocodeLocation()
+                        //}
+                        // TODO: manual update of coord, either properly implement or discard
+                        //Button("Set Location") {
+                        //    event.latitude = region.center.latitude
+                        //    event.longitude = region.center.longitude
+                        //    reverseGeocodeLocation()
+                        //}
+                        .padding(.top, 4)
+                        .frame(maxWidth: .infinity, alignment: .trailing)
+                    }
+                } header: {
+                    Text("Where?")
+                        .font(.headline)
+                        .foregroundColor(.black)
```

Comment:
use .primary rather than .black otherwise it's not visible in darkmode
```suggestion
                        .foregroundColor(.primary)
```